### PR TITLE
Add Hibernate integration page

### PR DESCRIPTION
**Pull request by vy**
This PR addresses #2511.

### CODE REVIEW

File: src/site/antora/modules/ROOT/examples/hibernate/log4j2.properties [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+logger.0.name = org.hibernate.SQL
+logger.0.level = DEBUG
+
+# Log JDBC bind parameters and extracted values
+logger.1.name = org.hibernate.type
+logger.1.level = TRACE
+logger.2.name = org.hibernate.orm.jdbc.bind
+logger.2.level = TRACE
+logger.3.name = org.hibernate.orm.jdbc.bind
+logger.3.level = TRACE
```
Comment (ppkarwasz) [inline src/site/antora/modules/ROOT/examples/hibernate/log4j2.properties:31]: I love that you also provided a `*.properties` file! :100:

File: src/site/antora/modules/ROOT/examples/hibernate/log4j2.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  <Loggers>
+
+    <!-- Log just the SQL -->
+    <Logger name="org.hibernate.SQL" level="DEBUG"/>
+
+    <!-- Log JDBC bind parameters and extracted values -->
+    <Logger name="org.hibernate.type" level="TRACE"/>
+    <Logger name="org.hibernate.orm.jdbc.bind" level="TRACE"/>
+    <Logger name="org.hibernate.orm.jdbc.extract" level="TRACE"/>
```
Comment (ppkarwasz) [inline src/site/antora/modules/ROOT/examples/hibernate/log4j2.xml:38]: I would keep this commented out with a warning:
> Be aware that JDBC statement parameters can contain sensitive data (passwords, credit card numbers, etc.).
> Uncomment this **only** in a dev environment (or start looking for a new job :wink:).
> See also "Arbiters" on how to switch config based on the environment.

**Review by ppkarwasz — CHANGES_REQUESTED** [review]
LGTM

I would just add a warning about logging JDBC parameters.

IMHO sensitive data should **never** be logged. In this case it is impossible to filter out sensitive data from what Hibernate logs, so users should complement Hibernate logs with manually written log statements that only use non-sensitive data.

File: src/site/antora/modules/ROOT/examples/hibernate/log4j2.properties (hunk 1 above)
Comment (vy) [reply src/site/antora/modules/ROOT/examples/hibernate/log4j2.properties:31]: Friends don't let friends suffer alone. :sunglasses:

File: src/site/antora/modules/ROOT/examples/hibernate/log4j2.xml (hunk 2 above)
Comment (vy) [reply src/site/antora/modules/ROOT/examples/hibernate/log4j2.xml]: Fixed in 2d86aa07c5997d912e0c5dcc0243b0b79e9b967e.

**Pull request merged**